Chatkit/Redux/Store/Store.swift

```diff
@@ -0,0 +1,110 @@
+
+enum Action {
+    case subscriptionEvent(Wire.Event.EventType)
+    case received(user: Wire.User)
+    case fetching(userWithIdentifier: String)
+}
+
+extension Action: Equatable {}
+
+protocol StoreDelegate: AnyObject {
+    func store(_ store: Store, didUpdateState state: State)
+}
+
+protocol HasStore {
+    var store: Store { get }
+}
+
+protocol Store {
+    var state: State { get }
+    func action(_ action: Action)
+}
+
+class ConcreteStore: Store {
+    
+    typealias Dependencies = Any // No dependencies for now
+    
+    private let dependencies: Dependencies
+    private weak var delegate: StoreDelegate?
+    
+    private(set) var state: State {
+        didSet {
+            if state != oldValue {
+                self.delegate?.store(self, didUpdateState: state)
+            }
+        }
+    }
+    
+    init(dependencies: Dependencies, delegate: StoreDelegate?) {
+        self.dependencies = dependencies
+        self.delegate = delegate
+        // Ensure the state is set *AFTER* the delegate so its `didSet` triggers a call to the delegate and its notified of the initial state
+        self.state = State.emptyState
+    }
+    
+    func action(_ action: Action) {
+        
+        let existingState = self.state
+        
+        let newState: State
+        
+        // TODO: this whole block needs to move into a `Reducer` or `ReductionManager`
+        // (and therefore this does not have full test coverage at present)
+        switch action {
+            
+        case let .subscriptionEvent(eventType):
```

Comment:
This might become hard to maintain as a single method. It's probably worth breaking this code into a few separate methods.

Reply:
See the comment a few lines above... 
```
        // TODO: this whole block needs to move into a `Reducer` or `ReductionManager`
        // (and therefore this does not have full test coverage at present)
```

It will undoubtly be split because each parsing block should end up in a dedicated `Reducer`.  This will form part of your work to implement `Reducers`.